Add Column Definition List for Output Columns for master_add_node

aamederen opened the pull request:
This change allows seeing the names of columns of `master_add_node`,
using `SELECT * FROM master_add_node(...)` by specifying output
columns in UDF definition.

Fixes #946

jasonmp85 commented:
Killed the `v6.0.0` tag to wait for this. Will `cherry-pick` onto `release-6.0` and re-tag (the tag was very new, hopefully no one gets too angry about rewriting it).

Ping me when merged; blocking release.

samay-sharma commented on src/backend/distributed/citus--6.0-17--6.0-18.sql:
```diff
@@ -0,0 +1,22 @@
+    RETURNS record                                                                            
+    LANGUAGE C STRICT                                                                         
+    AS 'MODULE_PATHNAME', $$master_add_node$$;
+COMMENT ON FUNCTION master_add_node(nodename text,                                            
```
(Minor) Could we collapse both arguments onto a single line? I tried that and it looks cleaner to me.

samay-sharma commented on src/backend/distributed/citus--6.0-17--6.0-18.sql:
```diff
@@ -0,0 +1,22 @@
+                                    nodeport integer)                                         
+    IS 'add node to the cluster';
+    
+RESET search_path;
```
I think Github needs a new line here.

samay-sharma commented on src/backend/distributed/citus--6.0-17--6.0-18.sql:
```diff
@@ -0,0 +1,22 @@
+
+SET search_path = 'pg_catalog';
+
+DROP FUNCTION master_add_node(nodename text, nodeport integer);
```
At a few places in the code, we use DROP FUNCTION IF EXISTS for similar purposes. Could we use that?

aamederen commented:
Hey @samay-sharma, good catches there. I have changed the commit and modified the regressions tests so that the `SELECT * FROM master_add_node(...)` usage is covered.

jasonmp85 commented:
@samay-sharma so is that a :shipit: or did you want a sec to check these test changes?

samay-sharma commented:
@aamederen: Looks to go me. :shipit:
